### PR TITLE
Fix loginPacket in serverTest.js for 1.17

### PR DESCRIPTION
**Pull request by piousdeer**
In 1.17 `previousGameMode` type was changed from `u8` to `i8`, setting it to 255 causes tests to fail.

### CODE REVIEW

**Comment by rom1504** [comment]
tests failing

**Comment by piousdeer** [comment]
Tests appeared to fail on the snapshot which you replaced with 1.17 in #866, should all work now.

File: test/serverTest.js [hunk 1]
```diff
@@ -222,6 +222,10 @@ for (const supportedVersion of mc.supportedVersions) {
           delete loginPacket.levelType
           delete loginPacket.difficulty
         }
+        if (version.version >= 755) { // 1.17
```
Comment (rom1504) [inline test/serverTest.js:225]: we only support the last version in examples

**Comment by rom1504** [comment]
> Tests appeared to fail on the snapshot which you replaced with 1.17 in #866, should all work now.

you will need to rebase to get the changes

**Comment by rom1504** [comment]
thanks, @u9g redid it in another pr

**Pull request closed without merging**